akka-actor-typed/src/main/scala/akka/actor/typed/internal/InternalMessage.scala

```diff
@@ -28,3 +29,10 @@ import akka.annotation.InternalApi
     with WrappedMessage {
   def adapt(): T = adapter(message)
 }
+
+/**
+ * INTERNAL API: Wrapped exception to pass a failure to adapt a message into the behavior stack and
+ * let supervision apply also to such failures.
+ */
+@InternalApi
+private[akka] final case class MessageAdaptionFailure(ex: Throwable) extends Signal
```

Comment:
To handle ExtensibleBehavior we can make this public and mention in the scaladoc of ExtensibleBehavior what it should do.

I don't see it as harmful if a user's behavior would be able to catch it. Might even be a nice power feature.

Reply:
Made it a public signal now